edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsUpload.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public interface IMetricsUpload
+    {
+        Task UploadAsync(IEnumerable<Metric> metrics, CancellationToken cancellationToken);
```

Comment:
does this upload in a transaction? if not, is it possible that one of metrics fail to upload? if so, how to report this failed metrics?

Reply:
It uploads them all at once. Currently, if it fails the metrics will be dropped. The desired behavior is one of the things we'll have to discuss next wednesday in the spec review.

Reply:
What is the final decision on this? please also update in spec.

Reply:
It will store the metrics for later and subscribe to a callback to be notified when connection is restored. This will be in a different pr.